packages/server/modules/core/errors/ratelimit.ts

```diff
@@ -0,0 +1,20 @@
+import { RateLimitBreached } from '@/modules/core/services/ratelimiter'
+import { BaseError } from '@/modules/shared/errors'
+import { Options } from 'verror'
+
+export class RateLimitError extends BaseError {
+  static defaultMessage =
+    'You have sent too many requests. You are being rate limited. Please try again later.'
+  static code = 'RATE_LIMIT_ERROR'
+
+  rateLimitBreached: RateLimitBreached
+
+  constructor(
+    rateLimitBreached: RateLimitBreached,
+    message?: string | null | undefined,
```

Comment:
I don't really see the point of allowing `null` here, if you don't want to add a message then skip the parameter, which is going to resolve to `undefined`. Additionally, I don't think you can put an optional parameter before more required parameters (message? before options).

```ts
constructor(
rateLimitBreached: RateLimitBreached,
message?: string,
options?: Options | Error
) {}
```